test/optimize_qdq_format_test.cpp

```diff
@@ -0,0 +1,198 @@
+                                      d6);
+        auto mb17 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280, 1, 1}}}), l13);
+        auto mb18 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280, 1, 1}}}), l3);
+        auto q3   = mm->add_instruction(migraphx::make_op("quantizelinear"), ap, mb17, mb18);
+        auto mb19 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280, 1, 1}}}), l13);
+        auto mb20 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280, 1, 1}}}), l3);
+        auto d7   = mm->add_instruction(migraphx::make_op("dequantizelinear"), q3, mb19, mb20);
+        auto rs   = mm->add_instruction(migraphx::make_op("reshape", {{"dims", {1, -1}}}), d7);
+        auto mb21 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280}}}), l12);
+        auto mb22 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280}}}), l2);
+        auto q4   = mm->add_instruction(migraphx::make_op("quantizelinear"), rs, mb21, mb22);
+        auto mb23 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280}}}), l12);
+        auto mb24 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1280}}}), l2);
+        auto d8 = mm->add_instruction(migraphx::make_op("dequantizelinear"), q4, mb23, mb24);
+        auto dot =
+            mm->add_instruction(migraphx::make_op("dot", {{"alpha", 1}, {"beta", 0}}), d8, d4);
+        auto mb25 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1000}}}), l10);
+        auto mb26 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1000}}}), l1);
+        auto q5   = mm->add_instruction(migraphx::make_op("quantizelinear"), dot, mb25, mb26);
+        auto mb27 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1000}}}), l10);
+        auto mb28 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1000}}}), l1);
+        auto d9   = mm->add_instruction(migraphx::make_op("dequantizelinear"), q5, mb27, mb28);
+        auto mb29 = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1000}}}), d3);
+        auto a2 = mm->add_instruction(migraphx::make_op("add"), d9, mb29);
+        mm->add_return({a2});
+
+        return p;
+    };
+
+    migraphx::program p1 = create_program();
+    migraphx::program p2 = create_program();
+
+    migraphx::optimize_qdq_format opt;
+    opt.apply(*p2.get_main_module());
+
+    match_find_qdq m1;
+    match_find_qdq m2;
+    migraphx::match::find_matches(*p1.get_main_module(), m1);
+    migraphx::match::find_matches(*p2.get_main_module(), m2);
```

Comment:
In the matcher.cpp test file there is a `find_match` function which should work better than trying to using the match finder.  This will let you pass the matcher directly:

```cpp
auto match_qdq = migraphx::match::name("dequantizelinear")(
                 migraphx::match::arg(0)(migraphx::match::name("quantizelinear")));
auto ins = find_match(mod, match_qdq);
```

You can copy this function here or just add it to matcher.hpp file so it can be reused across multiple files.